base_layer/core/src/transaction.rs

```diff
@@ -514,6 +514,32 @@ impl Transaction {
     pub fn calculate_ave_fee_per_gram(&self) -> f64 {
         (self.body.get_total_fee().0 as f64) / self.calculate_weight() as f64
     }
+
+    /// Returns the minimum maturity of the input UTXOs
+    pub fn min_input_maturity(&self) -> u64 {
+        let mut min_maturity = std::u64::MAX;
+        self.body
+            .inputs
+            .iter()
```

Comment:
self.body.inputs.iter().min() ?

Reply:
I would have to implement `Ord` for `TransactionInput` that orders them according to `features.maturity` to make it work for `iter().min()` or `iter().max()`. Im not sure this is the right way to order the inputs, but I was able to simplify this code by making use of fold: `       self.body.inputs.iter().fold(std::u64::MAX, |min_maturity, input| {
            min(min_maturity, input.features.maturity)
        })`